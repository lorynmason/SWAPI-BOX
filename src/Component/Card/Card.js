import React from 'react'
import '../styles/main.scss'
import PropTypes from 'prop-types'

const Card = ({ card, toggleFavorites, favorites }) => {
  const stats = Object.keys(card).map(key => {
    if (key === 'id') {
      return (
        <h3>{card[key]}</h3>
      )
    }
    return (
      <p>{key}: {card[key]}</p>
    )
  })
  let favoriteID = ''
  let favoriteBTN = ''
  if(favorites.includes(card.id)) {
    favoriteID = 'favorite-card'
    favoriteBTN = 'favorite-on'
  }
  return (
    <div className="card" key={card.id} id={favoriteID}>
      <button className="favorite-btn"  title="select favorite" id={favoriteBTN} type="button" onClick={() => toggleFavorites(card.id)}>
        <i className="fas fa-jedi" />
      </button>
      <div className="info">
        {stats}
      </div>
    </div>
  )
}

export default Card