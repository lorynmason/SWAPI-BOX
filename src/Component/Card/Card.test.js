import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card', () => {
  let wrapper
  const mockCard = {
    id: 'R2-D2',
    Homeworld: 'Naboo',
    Population: 4500000000,
    Species: 'Droi'
  }
  let mockFavorites = [{
    id: 'R2-D2'
  }]
  const mockFunc = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Card card={mockCard} favorites={mockFavorites} toggleFavorites={mockFunc}/>)
  })

  it('should match snapshot if it is a favorite',() => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot if it is not a favorite',() => {
    mockFavorites = [{
      id: 'C-3PO'
    }]
    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleFavorites when clicked',() => {
    wrapper.find('button').simulate('click');
    expect(mockFunc).toHaveBeenCalled()
  })
})