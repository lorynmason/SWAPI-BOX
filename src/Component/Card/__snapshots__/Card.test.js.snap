// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot if it is a favorite 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
      Object {
        "Homeworld": "Naboo",
        "Population": 4500000000,
        "Species": "Droi",
        "id": "R2-D2",
      }
    }
    favorites={
      Array [
        Object {
          "id": "R2-D2",
        },
      ]
    }
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "R2-D2",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-btn"
          id=""
          onClick={[Function]}
          title="select favorite"
          type="button"
        >
          <i
            className="fas fa-jedi"
          />
        </button>,
        <div
          className="info"
        >
          <h3>
            R2-D2
          </h3>
          <p>
            Homeworld
            : 
            Naboo
          </p>
          <p>
            Population
            : 
            4500000000
          </p>
          <p>
            Species
            : 
            Droi
          </p>
        </div>,
      ],
      "className": "card",
      "id": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-jedi"
          />,
          "className": "favorite-btn",
          "id": "",
          "onClick": [Function],
          "title": "select favorite",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-jedi",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              R2-D2
            </h3>,
            <p>
              Homeworld
              : 
              Naboo
            </p>,
            <p>
              Population
              : 
              4500000000
            </p>,
            <p>
              Species
              : 
              Droi
            </p>,
          ],
          "className": "info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "R2-D2",
            },
            "ref": null,
            "rendered": "R2-D2",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld",
                ": ",
                "Naboo",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld",
              ": ",
              "Naboo",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Population",
                ": ",
                4500000000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Population",
              ": ",
              4500000000,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species",
                ": ",
                "Droi",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species",
              ": ",
              "Droi",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "R2-D2",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-btn"
            id=""
            onClick={[Function]}
            title="select favorite"
            type="button"
          >
            <i
              className="fas fa-jedi"
            />
          </button>,
          <div
            className="info"
          >
            <h3>
              R2-D2
            </h3>
            <p>
              Homeworld
              : 
              Naboo
            </p>
            <p>
              Population
              : 
              4500000000
            </p>
            <p>
              Species
              : 
              Droi
            </p>
          </div>,
        ],
        "className": "card",
        "id": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-jedi"
            />,
            "className": "favorite-btn",
            "id": "",
            "onClick": [Function],
            "title": "select favorite",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-jedi",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                R2-D2
              </h3>,
              <p>
                Homeworld
                : 
                Naboo
              </p>,
              <p>
                Population
                : 
                4500000000
              </p>,
              <p>
                Species
                : 
                Droi
              </p>,
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "R2-D2",
              },
              "ref": null,
              "rendered": "R2-D2",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld",
                  ": ",
                  "Naboo",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld",
                ": ",
                "Naboo",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Population",
                  ": ",
                  4500000000,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Population",
                ": ",
                4500000000,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species",
                  ": ",
                  "Droi",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species",
                ": ",
                "Droi",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should match snapshot if it is not a favorite 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
      Object {
        "Homeworld": "Naboo",
        "Population": 4500000000,
        "Species": "Droi",
        "id": "R2-D2",
      }
    }
    favorites={
      Array [
        Object {
          "id": "R2-D2",
        },
      ]
    }
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "R2-D2",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-btn"
          id=""
          onClick={[Function]}
          title="select favorite"
          type="button"
        >
          <i
            className="fas fa-jedi"
          />
        </button>,
        <div
          className="info"
        >
          <h3>
            R2-D2
          </h3>
          <p>
            Homeworld
            : 
            Naboo
          </p>
          <p>
            Population
            : 
            4500000000
          </p>
          <p>
            Species
            : 
            Droi
          </p>
        </div>,
      ],
      "className": "card",
      "id": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-jedi"
          />,
          "className": "favorite-btn",
          "id": "",
          "onClick": [Function],
          "title": "select favorite",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-jedi",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              R2-D2
            </h3>,
            <p>
              Homeworld
              : 
              Naboo
            </p>,
            <p>
              Population
              : 
              4500000000
            </p>,
            <p>
              Species
              : 
              Droi
            </p>,
          ],
          "className": "info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "R2-D2",
            },
            "ref": null,
            "rendered": "R2-D2",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld",
                ": ",
                "Naboo",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld",
              ": ",
              "Naboo",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Population",
                ": ",
                4500000000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Population",
              ": ",
              4500000000,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species",
                ": ",
                "Droi",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species",
              ": ",
              "Droi",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "R2-D2",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-btn"
            id=""
            onClick={[Function]}
            title="select favorite"
            type="button"
          >
            <i
              className="fas fa-jedi"
            />
          </button>,
          <div
            className="info"
          >
            <h3>
              R2-D2
            </h3>
            <p>
              Homeworld
              : 
              Naboo
            </p>
            <p>
              Population
              : 
              4500000000
            </p>
            <p>
              Species
              : 
              Droi
            </p>
          </div>,
        ],
        "className": "card",
        "id": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-jedi"
            />,
            "className": "favorite-btn",
            "id": "",
            "onClick": [Function],
            "title": "select favorite",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-jedi",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                R2-D2
              </h3>,
              <p>
                Homeworld
                : 
                Naboo
              </p>,
              <p>
                Population
                : 
                4500000000
              </p>,
              <p>
                Species
                : 
                Droi
              </p>,
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "R2-D2",
              },
              "ref": null,
              "rendered": "R2-D2",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld",
                  ": ",
                  "Naboo",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld",
                ": ",
                "Naboo",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Population",
                  ": ",
                  4500000000,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Population",
                ": ",
                4500000000,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species",
                  ": ",
                  "Droi",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species",
                ": ",
                "Droi",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
